components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint/src/main/webapp/basicauth.jsp

```diff
@@ -160,7 +162,15 @@
                 identityMgtEndpointContext = IdentityUtil.getServerURL("/accountrecoveryendpoint", true, true);
             }
 
-            String url = identityMgtEndpointContext + "/recoverpassword.do?callback=" + Encode.forHtmlAttribute(urlEncodedURL);
+            HttpClient client = IdentityUtil.getHttpClientWithSSLSocketFactory(scheme, serverName);
+            String validationPointUrl = identityMgtEndpointContext + "/error.jsp";
+            HeadMethod head = new HeadMethod(validationPointUrl);
+            head.setFollowRedirects(false);
+
+            int status = client.executeMethod(head);
+            head.releaseConnection();
+            if (status == HttpURLConnection.HTTP_OK) {
+                String url = identityMgtEndpointContext + "/recoverpassword.do?callback=" + Encode.forHtmlAttribute(urlEncodedURL);
```

Comment:
Better to wrap  url creation with method which implies  its purpose

Reply:
Done